samples/java-spring-eventsourced-counter/src/it/java/com/example/CounterIntegrationTest.java

```diff
@@ -40,21 +40,21 @@ public class CounterIntegrationTest extends KalixIntegrationTestKitSupport { //
     // tag::test-topic[]
     @Autowired
     private KalixTestKit kalixTestKit; // <2>
-    private EventingTestKit.Topic commandsTopic;
-    private EventingTestKit.Topic eventsTopic;
+  private EventingTestKit.MockedSubscription commandsTopic;
```

Comment:
A Topic will be replaced by 2 new concepts:
1. `MockedSubscription` - user can inject msgs to subscription
2. `MockedDestination` - user can assert if msgs were produced (only `Publish.Topic` is supported for now)

Reply:
Also, are these names ok in the context of proto-first sdk?

Reply:
Do you mean if `Subscription` is a good name? I guess your doubt is related to the fact that in Protobuf SDKs, we don't use this word. Is that your point? 

What about `EventingTestKit.IncomingEvents` and `EventingTestKit.OutgoingEvents`?

We can maybe remove the "Mocked" prefix because it's already part of the TestKit. So maybe already clear enough that this is a test tool.

Reply:
For the Java SDK, the code has a Subscription to a stream of events. 

The TestKit simulates the incoming events being streamed to the method.

Reply:
> I guess your doubt is related to the fact that in Protobuf SDKs, we don't use this word. Is that your point?

exactly.

and then I would have:
 - getValueEntityIncomingEvents
 - getStreamIncomingEvents
 - getTopicOutgoingEvents
 
It's not bad. Except that `events` in terms of ValueEntity looks strange. I know it's the question of how we define an event... Maybe `Incoming/OutgoingMessages` ? 

Also, I've just noticed that the term `Stream` applies only to code-first, in proto we have `eventing.in.direct` :/

Reply:
Yes, I see. 

From one side it's bad that we use different namings and now that hit us back when writing common testkit code. I take responsibility for that.

I didn't want to repeat `eventing.in.direct` in the code-first API because I think it's quite cryptic for the users and it leaks internal details. The fact that is direct and not passing by a broker is not of interest. 

My thought was that `@Subscribe.Stream(service = "service-name", id = "stream-id")` was sufficient to express everything that a user needs to know about this subscription.

Reply:
Also the suffixing it with `Messages` sounds better, but we are not consistent anyway because the TestKit is called `EventingTestKit` and "to event" is not even a verb. 🤦

Reply:
I also prefer `Subscription`, `Stream` etc, and at the same time I understand the generality of `eventing`. The question is if we want to migrate `eventing` to a more specific naming at some point or not.